core/src/main/java/com/sequenceiq/cloudbreak/service/LoadBalancerConfigService.java

```diff
@@ -105,19 +105,21 @@ public String generateLoadBalancerEndpoint(Stack stack) {
     }
 
     public String getLoadBalancerUserFacingFQDN(Long stackId) {
-        Set<LoadBalancer> loadBalancers = loadBalancerPersistenceService.findByStackId(stackId);
-        if (!loadBalancers.isEmpty()) {
-            LoadBalancer preferredLB = loadBalancers.stream()
-                .filter(lb -> LoadBalancerType.PUBLIC.equals(lb.getType()))
-                .findAny()
-                .orElse(loadBalancers.iterator().next());
-            if (StringUtils.isNotEmpty(preferredLB.getFqdn())) {
-                return preferredLB.getFqdn();
-            } else {
-                return loadBalancers.stream()
-                    .map(LoadBalancer::getFqdn)
-                    .filter(StringUtils::isNotEmpty)
-                    .findFirst().orElse(null);
+        if (stackId != null) {
+            Set<LoadBalancer> loadBalancers = loadBalancerPersistenceService.findByStackId(stackId);
+            if (!loadBalancers.isEmpty()) {
+                LoadBalancer preferredLB = loadBalancers.stream()
+                    .filter(lb -> LoadBalancerType.PUBLIC.equals(lb.getType()))
+                    .findAny()
```

Comment:
why don't you filter out LBs without FQDN here and you won't need the next if-else